addons/web_editor/static/src/js/wysiwyg/widgets/link.js

```diff
@@ -298,6 +298,17 @@ export class Link extends Component {
         var allWhitespace = /\s+/gi;
         var allStartAndEndSpace = /^\s+|\s+$/gi;
         const isImage = this.props.link && this.props.link.querySelector('img');
+        let isDocument = false;
+        let directDownload = true;
+        try {
+            const parsedUrl = new URL(this.props.link.href);
+            
+            if (parsedUrl && parsedUrl.pathname.startsWith("/web/content/")) {
+                isDocument = true;
+                directDownload = parsedUrl.searchParams.get("download") === "true";
+            }
+        } catch {}
```

Comment:
Empty try catch triggers a warning in my editor config, so it's probably incorrect (I know I said it'd probably be fine but I guess not) 

I actually think that we should strip the domain from the URL that the media dialog returns, it would remove the "auto-convert relative link" since it would be relative from the start, and it would allow use to just do `this.props.link.href && this.props.link.href.startsWith("/web/content/")` without having to parse the URL. What do you think?